tracking/translations_parser/cli/taskcluster_group.py

```diff
@@ -95,109 +104,186 @@ def get_metrics_from_task(task: dict) -> list[Metric]:
         # Remove eventual slashes (e.g. <task_tag>-1/2) that cannot be written to the filesystem
         tag = MULTIPLE_TRAIN_SUFFIX.sub("", tag)
 
-        with tempfile.TemporaryDirectory() as d:
-            file = Path(d) / f"{tag}.txt"
-            with file.open("wb") as f:
-                f.write(log.tobytes())
-                f.flush()
-                metrics.append(Metric.from_file(Path(f.name), sep="-"))
+        with tempfile.TemporaryDirectory() as temp_dir:
+            file = Path(temp_dir) / f"{tag}.txt"
+            with file.open("wb") as log_file:
+                log_file.write(log.tobytes())
+                log_file.flush()
+                metrics.append(Metric.from_file(Path(log_file.name), sep="-"))
+
     return metrics
 
 
-def main() -> None:
-    args = get_args()
+def filter_task(task: dict) -> tuple[str, dict] | tuple[None, None]:
+    if task["status"]["state"] == "completed" and "vocab" not in task["task"]["tags"]["kind"]:    
+        name = task["task"]["tags"]["kind"]
+        prefix = name.split("-")[0]
+        if prefix == "train":
+            # Remove "train-" prefix from training task only to avoid duplicates
+            name = name[6:]
 
-    if args.loglevel:
-        logger.setLevel(args.loglevel)
+        # Teacher training may run multiple times (e.g. "-1/2" prefix)
+        suffix = ""
+        label = task["task"]["tags"].get("label")
+        if label and (re_match := MULTIPLE_TRAIN_SUFFIX.search(label)):
+            (suffix,) = re_match.groups()
+
+        task["name"] = name + suffix
+        return prefix, task
+    
+    return None, None
 
-    logger.info(f"Retrieving task group {args.group_id}")
-    # Ensure task group is readable
-    queue.getTaskGroup(args.group_id)
-    # Read project and experiment name
-    task_group = queue.task(args.group_id)
-    config = task_group["extra"]["action"]["context"]["input"]
-    experiment = config["experiment"]
-    project = f"{experiment['src']}-{experiment['trg']}"
-    group_name = f"{experiment['name']}_{args.group_id}"
 
-    logger.info(f"Listing completed tasks from group {args.group_id}")
-    resp = queue.listTaskGroup(args.group_id)
-    tasks = resp["tasks"]
-    continuation_token = resp.get("continuationToken")
+def get_training_tasks(group_id: str, grouped_tasks: dict[str, list[dict]]) -> list[list[dict]]:
```

Comment:
I would rename this to `list_training_tasks`. Same for above function

Reply:
Fixed in 0df72095f51f41de3e8d9434797a66dea560289f